### ISSUE TITLE
Assertion failed: `anchorBefore == null || anchorBefore.getTreeParent() == parent` in rule [kdoc-formatting]

### ISSUE THREAD

**Issue by petertrr** [saveourtool/diktat#1489]
## Describe the bug
```kotlin
package com.saveourtool.save.reporter.json

/**
 * Reporter that produces a JSON report as a [Report]
 *
 * @property out a sink for output
 *
 * @param builder additional configuration lambda for serializers module
 */
class JsonReporter(
    override val out: BufferedSink,
    builder: PolymorphicModuleBuilder<Plugin.TestFiles>.() -> Unit = {}
) : Reporter
```

## Observed behavior
Run `gradlew diktatFix` on this piece of code.
````
java.lang.AssertionError: Assertion failed: anchorBefore == null || anchorBefore.getTreeParent() == parent
	at org.jetbrains.kotlin.com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:57)
	at org.jetbrains.kotlin.com.intellij.openapi.diagnostic.Logger.error(Logger.java:187)
	at org.jetbrains.kotlin.com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:201)
	at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement.addChild(CompositeElement.java:571)
	at org.cqfn.diktat.ruleset.rules.chapter2.kdoc.KdocFormatting$checkBasicTagsOrder$2.invoke(KdocFormatting.kt:232)
	at org.cqfn.diktat.ruleset.rules.chapter2.kdoc.KdocFormatting$checkBasicTagsOrder$2.invoke(KdocFormatting.kt:217)
	at org.cqfn.diktat.ruleset.constants.Warnings.warnAndFix(Warnings.kt:266)
	at org.cqfn.diktat.ruleset.constants.Warnings.warnAndFix$default(Warnings.kt:208)
	at org.cqfn.diktat.ruleset.rules.chapter2.kdoc.KdocFormatting.checkBasicTagsOrder(KdocFormatting.kt:217)
...
````

## Environment information
* diktat version: 1.2.3
* build tool (maven/gradle): gradle

**Comment by 0x6675636b796f75676974687562** [comment]
Apparently, the problem was either present for almost as long as the check exists (ce7e8102c5671f3110142e4409bc0355f5164d73), or was caused by a somewhat later change (6843299f).
